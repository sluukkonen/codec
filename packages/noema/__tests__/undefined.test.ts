import { expectParseFailure, expectParseSuccess } from "./helpers.js"
import * as n from "../src/index.js"

describe("undefined", () => {
  test("should parse undefined", () => {
    expectParseSuccess(n.undefined, undefined)
  })

  test.each([
    [null, "invalid_undefined" as const],
    [0, "invalid_undefined" as const],
    ["noema", "invalid_undefined" as const],
    [{}, "invalid_undefined" as const],
    [[], "invalid_undefined" as const],
    [true, "invalid_undefined" as const],
    [false, "invalid_undefined" as const],
    [() => 1, "invalid_undefined" as const],
  ])("should not parse other values", (actual, code) => {
    expectParseFailure(n.undefined, actual, [{ code: code, path: [] }])
  })
})
