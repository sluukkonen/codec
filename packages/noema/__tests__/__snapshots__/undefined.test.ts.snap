// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined should not parse other values 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "undefined",
    "path": [],
    "received": "null",
  },
]
`;

exports[`undefined should not parse other values 2`] = `
[
  {
    "code": "invalid_type",
    "expected": "undefined",
    "path": [],
    "received": "string",
  },
]
`;
